Remove core security job (#90)
jenkins-infra/release: README.adoc

====

=== Security

> The security release follows the same process as the stable one except that artifacts are published in private. So we need promote git commits from a private repository to the public one then promote maven artifacts from a private maven repository to the public one.

Reviewer: ```suggestion
The security release follows the same process as the stable one except that artifacts are published in private. So we need to promote git commits from a private repository to the public one then promote maven artifacts from a private maven repository to the public one.
```

No sentence-per-line formatting 😦

> The security release follows the same process as the stable one excepted that artifacts are published in private location. So we need promote git commits from a private repository to the public one then promotes maven artifacts from a private maven repository to the public one.

Reviewer 2: ```suggestion
The security release follows the same process as the stable one except that artifacts are published in private. So we need promote git commits from a private repository to the public one then promote maven artifacts from a private maven repository to the public one.
```

Author: Thanks, one day I'll be good at writing (english) documentation

. Prepare jenkinsci-cert/jenkins repository -> missing documentation link

> . Create a branch on jenkins-infra/release with a branch name that match the release branch from jenkinsci/jenkins like `security-stable-2.135`

Reviewer 2: ```suggestion
. Create a branch on jenkins-infra/release with a branch name that match the release branch from jenkins-cert/jenkins like `security-stable-2.135`
```

. Review and update the security environment https://github.com/jenkins-infra/release/blob/master/profile.d/security[file] with:
.. `RELEASE_GIT_BRANCH` set to the `jenkinsci-cert/jenkins` release branch like `security-stable-2.135`
.. `JENKINS_VERSION` set to the final release version that will be packaged. If set to 'stable' or 'weekly' then the packaging job will try to guess the version based on what was pushed to the maven repository. cfr settings.
.. `PACKAGING_GIT_BRANCH` set to the appropriated jenkinsci/packaging branch
.. `MAVEN_REPOSITORY_NAME` set to the maven repository name where we are going to publish staging maven artifacts. This is also the source location used by the packaging job to build distribution packages
. Trigger the generic Release link:https://release.ci.jenkins.io/job/core/job/release/[job] from the appropriated branch like `security-stable-2.235`
.. Force repository scan

> .. Trigger the first build to have access to job parameter

Reviewer: ```suggestion
.. Trigger the first build to have access to job parameter and immediately abort it
```

.. Trigger a job with the correct parameters
... `RELEASE_PROFILE` set to `security`
... `RELEASE_GIT_BRANCH` set to `unused` as we already define it in the release profile file, which overrides the job parameter
... `MAVEN_REPOSITORY_NAME` set to `unused` as we already define it in the release profile file, which overrides the job parameter
... `VALIDATION_ENABLED` set to true in to have a validation stage

> .. Trigger the generic Packaging job link:https://release.ci.jenkins.io/job/core/job/package/[job] from the appropriated branch like `security-stable-2.235` with correct parameters

Reviewer 2: ```suggestion
.. Trigger the generic Packaging job link:https://release.ci.jenkins.io/job/core/job/package/[job] from the appropriate branch like `security-stable-2.235` with correct parameters
```
